src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs

```diff
@@ -42,6 +42,10 @@ public static IEnumerable<object[]> SerializableEqualityComparers_MemberData()
         /// </summary>
         public static IEnumerable<object[]> SerializableObjects_MemberData()
         {
+            // Save old culture and set a fixed culture for object instantiation
+            CultureInfo oldCulture = CultureInfo.CurrentCulture;
+            CultureInfo.CurrentCulture = new CultureInfo("en-US");
```

Comment:
@stephentoub I will follow your suggestion and change the method from an iterator to simply return the object[][] array and inside of it use a try - finally block. Thanks for the tip, that's better than my current solution.

I don't want to change the test roundtripping test to use RemoteExecutor as some downsides come with it:
a) Test execution time increases dramatically
b) Refactoring would needed to be done so that all the object instantiations and checks would happen inside the RemoteExecutor. That would mean that we can't use the MemberData xunit pattern and everything would execute in one test instead of 300 separate tests. For mc logging I don't really want to go that way, mainly, because I already had it like that before and even with extensive logging the test behavior and possible exceptions was hard to understand.

Because of these reasons I want to avoid using an RemoteExecutor here. If other tests in other classes rely on the culture I could still put the xunit argument (forgot its name) onto the class to not execute both at the same time.